src/stack_limiter/mod.rs

```diff
@@ -194,22 +246,36 @@ fn compute_stack_cost(func_idx: u32, module: &elements::Module) -> Result<u32, &
 			locals_count.checked_add(local_group.count()).ok_or("Overflow in local count")?;
 	}
 
-	let max_stack_height = max_height::compute(defined_func_idx, module)?;
+	let max_stack_height = MaxStackHeightCounter::new_with_context(context, injection_fn)
+		.count_instrumented_calls(true)
+		.compute_for_defined_func(defined_func_idx)?;
 
 	locals_count
 		.checked_add(max_stack_height)
 		.ok_or("Overflow in adding locals_count and max_stack_height")
 }
 
-fn instrument_functions(
+fn instrument_functions<I: IntoIterator<Item = Instruction>>(
 	ctx: &mut Context,
 	module: &mut elements::Module,
-) -> Result<(), &'static str> {
+	injection_fn: impl Fn(&FunctionType) -> I,
+) -> Result<(), &'static str>
+where
+	I::IntoIter: ExactSizeIterator + Clone,
+{
+	let types = module.type_section().map(|ts| ts.types()).unwrap_or(&[]).to_vec();
+	let functions = module.function_section().map(|fs| fs.entries()).unwrap_or(&[]).to_vec();
```

Comment:
are you sure that's correct? On that level module must have types and functions sections. Better return an error here, I guess.

Reply:
isn't is was discussed here? https://github.com/gear-tech/wasm-instrument/pull/5#discussion_r1370069875

Reply:
done in e41fe549c04a5e8610f06bf94271849b49e6fcbb